producer/producer_sf.go

```diff
@@ -196,13 +196,19 @@ func ParseSampledHeaderConfig(flowMessage *flowmessage.FlowMessage, sampledHeade
 				identificationEncap = identification
 				fragOffsetEncap = fragOffset
 				flowLabelEncap = flowLabel
+				nextHeader, tos, ttl, identification, fragOffset, flowLabel = 0, 0, 0, 0, 0, 0
 
 				if (etherType[0] == 0x8 && etherType[1] == 0x0) ||
 					(etherType[0] == 0x86 && etherType[1] == 0xdd) {
 					srcIPEncap = srcIP
 					dstIPEncap = dstIP
 					encap = true
 				}
+				if etherType[0] == 0x88 && etherType[1] == 0x0b { // PPP
+					srcIPEncap = srcIP
+					dstIPEncap = dstIP
+					srcIP, dstIP = nil, nil
```

Comment:
Would be missing encap.
If there are no other src/dst IP, we should not use encap

Reply:
Sorry what I'm trying to fix is parsing PPTP (PPP over GRE) packet samples, so I've updated the PR title. 

> Could you provide an sFlow or PPP packet capture for testing?

Can you try with the capture file [here](https://packetlife.net/captures/protocol/pptp/)?
PPP over GRE over IP packets follow after the TCP handshake. 

> Would be missing encap.
> If there are no other src/dst IP, we should not use encap

srcIP and dstIP need to be copied to srcIPEncap and dstIPEncap respectively here as long as we are leaving `for encap && iterations <= 1 {` loop, because they will be swapped again in [this following code](https://github.com/cloudflare/goflow/blob/master/producer/producer_sf.go#L214-L219). 

> Is there a need to reset those variables?

Yes for the same reason as above, again as long as we are leaving the `for` block.

Reply:
For instance, without my patch the packet No. 6 of https://www.cloudshark.org/captures/7a6644ad437e would be parsed as: 

`{
        "DstAddr": "<nil>",
        "DstAddrEncap": "20.0.0.1",
        "DstPort": 0,
        "Etype": 2048,
        "EtypeEncap": 34827,
        "HasEncap": true,
        "IPTTL": 64,
        "IPTTLEncap": 64,
        "IPTos": 0,
        "IPTosEncap": 0,
        "Proto": 47,
        "ProtoEncap": 47,
        "SrcAddr": "<nil>",
        "SrcAddrEncap": "20.0.0.2",
}
`
With my patch it would end up like so. 
`{
        "DstAddr": "20.0.0.1",
        "DstAddrEncap": "<nil>",
        "DstPort": 0,
        "Etype": 2048,
        "EtypeEncap": 34827,
        "HasEncap": true,
        "IPTTL": 64,
        "IPTTLEncap": 0,
        "IPTos": 0,
        "IPTosEncap": 0,
        "Proto": 47,
        "ProtoEncap": 0,
        "SrcAddr": "20.0.0.2",
        "SrcAddrEncap": "<nil>",
}
`

Reply:
And even now I believe `Proto` and `ProtoEncap` should be vice versa. Please check #52.